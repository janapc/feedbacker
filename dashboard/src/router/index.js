import { createRouter, createWebHistory } from 'vue-router';

const Home = () => import('../views/Home/index.vue');
const Feedbacks = () => import('../views/Feedbacks/index.vue');
const Credencials = () => import('../views/Credencials/index.vue');

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home,
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true,
    },
  },
  {
    path: '/credencials',
    name: 'Credencials',
    component: Credencials,
    meta: {
      hasAuth: true,
    },
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' },
  },
];

const router = createRouter({
  history: createWebHistory('/'),
  routes,
});

export default router;
