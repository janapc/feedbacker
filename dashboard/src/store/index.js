import { readonly } from 'vue';
import UserModule from './user';
import GlobalModule from './global';

export default readonly({
  User: UserModule,
  Global: GlobalModule,
});
