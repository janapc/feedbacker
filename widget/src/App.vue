<template>
  <widget />
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import Widget from './views/Widget/index.vue';

export default defineComponent({
  components: { Widget }
});
</script>
