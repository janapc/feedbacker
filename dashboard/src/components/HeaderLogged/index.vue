<template>
  <div class="flex items-center justify-between w-4/5 max-w-6xl py10">
    <div class="w-28 lg:w-36">
      <img class="w-full" src="../../assets/images/logo_white.png" alt="logo" />
    </div>

    <div class="flex">
      <ul class="flex list-none">
        <li
        id="router-credencials"
          @click="changeRouter('Credencials')"
          class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"
        >
          Credencias
        </li>

        <li
          @click="changeRouter('Feedbacks')"
          class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"
        >
          Feedbacks
        </li>

        <li id="logout-button" @click="handleLogout" class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none">{{ logoutLabel }}</li>
      </ul>
    </div>
  </div>
</template>

<script>
import { computed } from 'vue';
import { useRouter } from 'vue-router';

import useStore from '../../hooks/useStore';
import { cleanCurrentUser } from '../../store/user';

export default {
  setup () {
    const router = useRouter();
    const store = useStore('User');

    const logoutLabel = computed(() => {
      if (!store.currentUser.name) {
        return '...';
      }

      return `${store.currentUser.name} (sair)`;
    });

    function changeRouter (name) {
      router.push({ name });
    }

    function handleLogout () {
      window.localStorage.removeItem('token');
      cleanCurrentUser();
      router.push(
        { name: 'Home' }
      );
    }

    return {
      changeRouter,
      logoutLabel,
      handleLogout
    };
  }
};
</script>

<style></style>
