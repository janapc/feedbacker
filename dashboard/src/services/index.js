import axios from 'axios';

import router from '../router';
import AuthService from './auth';
import UsersService from './users';
import FeedbacksService from './feedbacks';

import { setGlobalLoading } from '../store/global';

const API_ENVS = {
  production: 'https://backend-vue3js-ixvpypm03.vercel.app/',
  development: 'http://localhost:3000',
  local: 'http://localhost:3000',
};

const httpClient = axios.create({
  baseURL: API_ENVS[process.env.NODE_ENV] ?? API_ENVS.local,
});

httpClient.interceptors.response.use(
  (response) => {
    setGlobalLoading(false);
    return response;
  },
  (error) => {
    const canThrowAnError = error.request.status === 0 || error.request.status === 500;

    if (canThrowAnError) {
      setGlobalLoading(false);
      throw new Error(error.message);
    }
    if (error.response.status === 401) {
      router.push({ name: 'Home' });
    }
    setGlobalLoading(false);
    return error;
  },
);

httpClient.interceptors.request.use((config) => {
  setGlobalLoading(true);
  const token = window.localStorage.getItem('token');

  if (token) {
    config.headers.common.Authorization = `Bearer ${token}`;
  }
  return config;
});

export default {
  auth: AuthService(httpClient),
  users: UsersService(httpClient),
  feedbacks: FeedbacksService(httpClient),
};
