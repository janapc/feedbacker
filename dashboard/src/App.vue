<template>
  <modal-factory />
  <router-view />
</template>

<script>
import { watch } from 'vue';
import { useRoute, useRouter } from 'vue-router';

import ModalFactory from './components/ModalFactory/index.vue';
import services from './services';
import { setCurrentUser } from './store/user';

export default {
  components: { ModalFactory },
  setup() {
    const route = useRoute();
    const router = useRouter();

    watch(
      () => route.path,
      async () => {
        if (route.meta.hasAuth) {
          const token = window.localStorage.getItem('token');

          if (!token) {
            router.push({ name: 'Home' });
          }

          const { data } = await services.users.getMe();

          setCurrentUser(data);
        }
      },
    );
  },
};
</script>
