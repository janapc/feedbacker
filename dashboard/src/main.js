import { createApp } from 'vue';
import Toast, { POSITION } from 'vue-toastification';
import App from './App.vue';
import router from './router';

import 'animate.css';

import '@/assets/css/tailwind.css';
import '@/assets/css/fonts.css';
import 'vue-toastification/dist/index.css';

const app = createApp(App);
app.use(router);
app.use(Toast, { position: POSITION.BOTTOM_RIGHT });
app.mount('#app');
