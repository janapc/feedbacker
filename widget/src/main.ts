import { createApp } from 'vue';

import { setup } from './utils/boostrap';

import App from './App.vue';
import Playground from './views/Playground/index.vue';

import './assets/css/fonts.css';
import './assets/css/tailwind.css';

import 'animate.css';

setup({
  onDevelopment: () => {
    createApp(Playground).mount('#app');
  },
  onProduction: () => {
    createApp(App).mount('#app');
  }
});
