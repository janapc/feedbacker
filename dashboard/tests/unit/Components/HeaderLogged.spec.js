import { shallowMount, flushPromises } from '@vue/test-utils';
import { createRouter, createWebHistory } from 'vue-router';

import HeaderLogged from '../../../src/components/HeaderLogged/index.vue';
import { routes } from '../../../src/router';

const router = createRouter({
  history: createWebHistory('/'),
  routes,
});

const localStorageMock = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
};
global.localStorage = localStorageMock;

const mockStore = { currentUser: {} };
jest.mock('../../../src/hooks/useStore', () => () => mockStore);

describe('<HeaderLogged />', () => {
  it('should render header logged correctly', async () => {
    router.push('/');
    await router.isReady();
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render 3 dots when there\'s not user logged', async () => {
    router.push('/');
    await router.isReady();
    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    const btnLogout = wrapper.find('#logout-button');
    expect(btnLogout.text()).toBe('...');
  });

  it('should render user name when there\'s not user logged', async () => {
    router.push('/');
    await router.isReady();
    mockStore.currentUser.name = 'Banana';

    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    const btnLogout = wrapper.find('#logout-button');
    expect(btnLogout.text()).toBe('Banana (sair)');
  });

  it('should log out of account', async () => {
    localStorage.setItem('token', '123');

    router.push('/');
    await router.isReady();
    mockStore.currentUser.name = 'Banana';

    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    const btnLogout = wrapper.find('#logout-button');
    expect(btnLogout.text()).toBe('Banana (sair)');

    await btnLogout.trigger('click');
    mockStore.currentUser = {};

    await flushPromises();

    expect(localStorage.getItem('token')).toBe(null);
    expect(router.currentRoute.value.name).toBe('Home');
  });

  it('should change of route', async () => {
    router.push('/');
    await router.isReady();
    mockStore.currentUser.name = 'Banana';

    const wrapper = shallowMount(HeaderLogged, {
      global: {
        plugins: [router],
      },
    });

    const btnCredencials = wrapper.find('#router-credencials');
    expect(btnCredencials.text()).toBe('Credencias');

    await btnCredencials.trigger('click');

    await flushPromises();

    expect(router.currentRoute.value.name).toBe('Credencials');
  });
});
