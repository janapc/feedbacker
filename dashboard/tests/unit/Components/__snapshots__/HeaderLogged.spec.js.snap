// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> should render header logged correctly 1`] = `
<div class="flex items-center justify-between w-4/5 max-w-6xl py10">
  <div class="w-28 lg:w-36"><img class="w-full" src="" alt="logo"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li id="router-credencials" class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Credencias </li>
      <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"> Feedbacks </li>
      <li id="logout-button" class="px-6 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none">...</li>
    </ul>
  </div>
</div>
`;
